File: Data Structures & Algorithms/reorder-linked-list/submission-0.java
class Solution {
    public void reorderList(ListNode head) {
        List<ListNode> nodes = new ArrayList<>();
        ListNode curr = head;
        while (curr != null) {
            nodes.add(curr);
            curr = curr.next;
        }

        int p1 = 0;
        int p2 = nodes.size() - 1;

        while (p1 < p2) {
            nodes.get(p1).next = nodes.get(p2);  // left → right
            p1++;

            if (p1 == p2) break;                  // odd-length guard

            nodes.get(p2).next = nodes.get(p1);  // right → new left
            p2--;
        }

        nodes.get(p1).next = null;                // null-terminate
    }
}

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        nodes = []
        curr = head
        while curr:
            nodes.append(curr)
            curr = curr.next

        l = 0
        r = len(nodes) - 1

        while l < r:
            if nodes[l].val != nodes[r].val:
                return False
            l+=1
            r-=1

        return True  
